kratos/factories/standard_linear_solver_factory.cpp

```diff
@@ -31,55 +29,81 @@
 #include "linear_solvers/skyline_lu_custom_scalar_solver.h"
 #include "spaces/ublas_space.h"
 
-namespace Kratos
+namespace Kratos {
+
+
+namespace detail {
+
+
+template <class TSparseDataType,
+          class TDenseDataType>
+void RegisterLinearSolvers()
 {
-    void RegisterLinearSolvers()
-    {
-        using SpaceType = TUblasSparseSpace<double>;
-        using LocalSpaceType = TUblasDenseSpace<double>;
-        using ComplexSpaceType = TUblasSparseSpace<std::complex<double>>;
-        using ComplexLocalSpaceType = TUblasDenseSpace<std::complex<double>>;
-        // using LinearSolverType = LinearSolver<SpaceType,  LocalSpaceType>;
-        // using IterativeSolverType = IterativeSolver<SpaceType,  LocalSpaceType>;
-        using CGSolverType = CGSolver<SpaceType,  LocalSpaceType>;
+    using SpaceType = TUblasSparseSpace<TSparseDataType>;
```

Comment:
BTW, we should eventually replace the linear solver factory with the register @KratosMultiphysics/technical-committee I don't know if a good moment to do that

Reply:
I was thinking about that too, but other parts of Kratos still expect linear solvers to be in `KratosComponents` and I didn't want to change too many unrelated things at once.

Reply:
A priori we can define in both database and add a warning in the factory, opinion @KratosMultiphysics/technical-committee ?